[DOCS] Add simulate ref (#58579)
elastic/elasticsearch: docs/reference/indices/simulate-index.asciidoc

[[indices-simulate-index]]
=== Simulate index API
++++
<titleabbrev>Simulate index</titleabbrev>
++++

experimental[]

Returns the index settings that would be applied to the specified index from an 
<<indices-templates, index template>>.

////
[source,console]
--------------------------------------------------
PUT _index_template/template_1
{
  "index_patterns": ["myindex*"],
  "template": {
    "settings": {
      "number_of_shards": 1
    }
  }
}
--------------------------------------------------
// TESTSETUP

[source,console]
--------------------------------------------------
DELETE _index_template/*
--------------------------------------------------
// TEARDOWN
////

[source,console]
--------------------------------------------------
POST /_index_template/_simulate_index/myindex-1
--------------------------------------------------

[[simulate-index-api-request]]
==== {api-request-title}

`POST /_index_template/_simulate_index/<index>`

[[simulate-index-api-path-params]]
==== {api-path-parms-title}

`<index>`::
(Required, string)
The name of the index to simulate. 

[[simulate-index-api-query-params]]
==== {api-query-parms-title}

`cause`::

> (Optional, string) The reason for running the simulation. 

Author: Yeah, that seems like a bug. I'll remove it and open an issue.

include::{es-ref-dir}/rest-api/common-parms.asciidoc[tag=master-timeout]

[[simulate-index-api-response-body]]
==== {api-response-body-title}

> `template`::

Reviewer: ```suggestion
`template`::
(object)
```

+
.Properties of `template`
[%collapsible%open]
====
include::{es-ref-dir}/rest-api/common-parms.asciidoc[tag=aliases]

include::{es-ref-dir}/rest-api/common-parms.asciidoc[tag=mappings]

include::{es-ref-dir}/rest-api/common-parms.asciidoc[tag=settings]
====

[[simulate-index-api-example]]
==== {api-examples-title}

The following example shows the settings that would be applied to `logdata-2019-02-01` by 
an existing template.

[source,console]
--------------------------------------------------
PUT /_component_template/ct1                             <1>
{
  "template": {
    "settings": {
      "index.number_of_shards": 2
    }
  }
}

PUT /_component_template/ct2                             <2>
{
  "template": {
    "settings": {
      "index.number_of_replicas": 0
    },
    "mappings": {
      "properties": {
        "@timestamp": {
          "type": "date"
        }
      }
    }
  }
}

PUT /_index_template/final-template                      <3>
{
  "index_patterns": ["logdata-*"],
  "composed_of": ["ct1", "ct2"],
  "priority": 5
}

POST /_index_template/_simulate_index/logdata-2020-02-02 <4>
--------------------------------------------------

> <1> Creating a component template (ct1) setting the number of shards to two
> <2> Creating another component template (ct2) setting the number of replicas to zero with mappings
> <3> Creating an index template called "final" template using ct1 and ct2
> <4> Show the settings that would be applied to the `logdata-2019-02-01`index

Reviewer: Rewords. Also fixes the index name to match the example.
```suggestion
<1> Creates a component template (`ct1`) that sets the number of shards to two
<2> Creates another component template (`ct2`) that sets the number of replicas to zero with mappings
<3> Creates an index template called `final-template` using the `ct1` and `ct2` component templates
<4> Show the settings that would be applied to the `logdata-2020-02-02`index
```

> The response shows the index settings, mappings, and aliases applied by the specified template:

Reviewer: Technically, the template isn't specified here. I reworded, but you could say something like "template matching the index name."
```suggestion
The response shows the index settings, mappings, and aliases applied by the `final-template` template:
```

[source,console-result]
---------------------------------------------------------
{
  "template" : {
    "settings" : {
      "index" : {
        "number_of_shards" : "2",
        "number_of_replicas" : "0" <1>
      }
    },
    "mappings" : {
      "properties" : {
        "@timestamp" : {
          "type" : "date"          <2>
        }
      }
    },
    "aliases" : { }
  },
  "overlapping" : [ ]
}
---------------------------------------------------------
<1> The number of replicas from the simulated template body

> <2> The `@timestamp` field inherited from the "ct2" component template

Reviewer: I'm not sure these callouts are needed. If you keep them, I'd remove the first article (`The`).
